### ISSUE TITLE
Couldn't make a guess for

### ISSUE THREAD

**Issue by brozikcz** [square/dagger#534]
I get this error:

```
An annotation processor threw an uncaught exception.
Consult the following stack trace for details.
java.lang.IllegalArgumentException: couldn't make a guess for CollectionAdapter<cz.babel.babelapp.core.models.MessageAndAttachment>
    at com.squareup.javapoet.Util.checkArgument(Util.java:64)
    at com.squareup.javapoet.ClassName.bestGuess(ClassName.java:159)
    at dagger.internal.codegen.Util$2.visitError(Util.java:199)
    at dagger.internal.codegen.Util$2.visitError(Util.java:183)
    at com.sun.tools.javac.code.Type$ErrorType.accept(Type.java:1837)
    at dagger.internal.codegen.Util.injectableType(Util.java:183)
    at dagger.internal.codegen.Util.bindingOf(Util.java:396)
```

for class: `public class MessageCollectionAdapter extends CollectionAdapter<MessageAndAttachment> {`

This problem is after upgrade from 1.2.2 to 1.2.3. With 1.2.2 works fine.

What its wrong? Thx.

**Comment by swankjesse** [comment]
Take a look at ParameterizedTypeName.

**Comment by tbroyer** [comment]
@swankjesse I think you mixed up projects 😋

**Comment by brozikcz** [comment]
Thanks for answer, but I don't understand. What do you mean about ParameterizedTypeName?

**Comment by brozikcz** [comment]
Problem solved, my bug in mix dagger 1.2.2 from own lib and newest version 1.2.5 from main project.
